dist/index.html

```diff
@@ -35,18 +36,77 @@
         })
       }
     </script>
+
     <link rel="icon" type="image/png" href="favicon-32x32.png">
   </head>
+
   <body>
     <div class="content" style="position: absolute; width: 100vw; height: 100vh; top: 0; left: 0; overflow: hidden; background: black; margin: 0; padding: 0;"></div>
     <script type="text/javascript" src="itkVtkViewer.js">
     </script>
     <script>
-      var container = document.querySelector('.content');
-      var viewer = itkVtkViewer.processURLParameters(container, { fullscreen: true });
+     async function setupViewerForImJoy(container) {
+      const imjoyRPC = await imjoyLoader.loadImJoyRPC({
+        api_version: '0.2.3'
+      })
+      const version = itkVtkViewer.version
+      const api = await imjoyRPC.setupRPC({
+        name: 'itk-vtk-viewer',
+        version,
+        description: '2D / 3D web image, mesh, and point set viewer using itk.js and vtk.js ',
+        type: 'rpc-window',
+        docs: 'https://kitware.github.io/itk-vtk-viewer/docs/imjoy.html',
+      })
+      api.export({
+        setup() {
+          this.viewer = null
+          itkVtkViewer.createViewerFromLocalFiles(container).then((viewer) => {
+            this.viewer = viewer
+          })
+          api.log("itk-vtk-viewer loaded successfully.")
+        },
+
+        async run(ctx) {
+          if (ctx.data && ctx.data.image) {
+            await this.setImage(ctx.data.image)
+          }
+        },
+
+        async setImage(image) {
+          let itkImage = image
+          if (image.shape !== undefined && image.stride !== undefined) {
+            itkImage = itkVtkViewer.utils.ndarrayToItkImage(image)
+          }
+
+          const vtkImage = itkVtkViewer.utils.vtkITKHelper.convertItkToVtkImage(itkImage)
+          const is2D = itkImage.imageType.dimension === 2
+          if (this.viewer === null) {
+            this.viewer = await itkVtkViewer.createViewer(container, {
+              image: vtkImage,
+              pointSets: null,
+              geometries: null,
+              use2D: is2D,
+              rotate: false
+            })
+          } else {
+            await this.viewer.setImage(vtkImage)
```

Comment:
Will this be an issue if we pass another image with different size, dimension (2D vs 3D) to the viewer? If not we might need to check if the new image has the same dimension as the last one.

Reply:
That is likely a corner use case, but, yes, the behavior may not be ideal. However, I am starting some major refactoring where we will address this.